crates/containerd-shim-wasm/src/container/context.rs

```diff
@@ -44,6 +47,24 @@ pub enum Source<'a> {
     Oci(&'a [WasmLayer]),
 }
 
+impl<'a> Source<'a> {
+    pub fn as_bytes(&self) -> anyhow::Result<Cow<'a, [u8]>> {
+        match self {
+            Source::File(path) => {
+                let path = path
+                    .resolve_in_path_or_cwd()
+                    .next()
+                    .context("module not found")?;
+                Ok(Cow::Owned(std::fs::read(path)?))
```

Comment:
**thought:** We optimize for no copy in the `Oci` case by using `Cow`, but we read the file every time in the `File` case. Might be premature optimiation, but the file content could be cached in memory here to avoid multiple reads.
I'm fine with leaving it as it is because the unoptimized `File` case would be the _legacy_ workflow. Also we could revisit if we ever observe any perforance bottleneck here.